core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -641,6 +641,10 @@ private void readCurrentBlockToPos(long pos) throws IOException {
     if (len <= 0) {
       return;
     }
+    if (mPos % mBlockSize == 0 && pos - mPos >= mBlockSize) {
+      closeOrCancelCacheStream();
```

Comment:
The cache out stream is not null. It needs to be cancelled. Otherwise the check in line 608 fails